design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,176 @@
+
+## Goals
+
+This design document should:
+
+* Define the steps for deploying a ClusterLink gateway to a K8s cluster.
+* Define the task functions of the ClusterLink deployment operator for setting up the ClusterLink components in the cluster.
+* Define the CRD (Custom Resource Definition) for the ClusterLink deployment operator.
+
+## Non-Goals
+
+The following aspects are explicitly excluded and are out of scope in the current design:
+
+* This document focuses only on the deployment of ClusterLink to the K8s cluster.
+The deployment to another environment as VMs is out of scope.
+* Security considerations related to the creation of certificates for the fabric or each site by the fabric administrator are also beyond the scope of this document.
+
+## Proposal
+
+The proposal describes how the ClusterLink project can be deployed on any K8s Cluster.
+In this proposal, we distinguish between two entities:
+
+1. Fabric administrator: manages the fabric and is responsible for determining which clusters can join the fabric.
+1. Site administrator: manages the k8s Cluster and is responsible for deploying the ClusterLink into the cluster.
+
+Before deploying the ClusterLink, a few prerequisite steps should be completed:
+
+1. The fabric administrator should create the site certificates (public and private) and transfer them, along with the public CA certificate, to the site administrator.
+2. The site administrator should deploy the certificates as k8s secrets to the cluster.
+3. The site administrator should deploy the ClusterLink operator to the cluster and register ClusterLink Custom Resource Definition (CRD) class to the cluster.
+
+After completing all the prerequisites, the site administrator can deploy the ClusterLink CRD instance to the operator. The operator will then create the ClusterLink components, including:
+
+* A ClusterLink dedicated namespace, which by default will be cl-operator-ns.
+* Deployment of ClusterLink controlplane components, including controlplane-pod, controlplane-service, and RBAC roles.
+* Deployment of ClusterLink dataplane components, including dataplane pods (single or multiple) and the dataplane-service.
+* Deployment of ClusterLink ingress, providing an external access point using load-balancer/node-port/gateway-API service."
+
+Overall, the ClusterLink deployment stages are:
+
+<img src="deployment.png" width="800" height="400" alt="ClusterLink Deployment Stages"/>
+
+The ClusterLink operator will have privileged permissions, allowing it to create a dedicated namespace and the ClusterLink components within.
+The ClusterLink components will be created in `clusterlink-system-ns`, and the control-plane will have privileged permissions within the namespace.
```

Comment:
Actually the control plane needs to watch objects in other namespaces as well. 
Also, why does it need to have privileges on its own namespace?